unit test flake: controller/persistentvolume

Not many details in the logs:
`
FAIL	k8s.io/kubernetes/pkg/controller/persistentvolume	2.721s
`

https://console.cloud.google.com/storage/browser/kubernetes-jenkins/pr-logs/pull/25916/kubernetes-pull-test-unit-integration/26923/

```
--- FAIL: TestControllerSync (0.15s)
	framework_test.go:711: Test "5-2 - complete bind": Claim check failed [A-expected, B-got result]: {"claim5-2":{"metadata":{"name":"claim5-2","namespace":"default","selfLink":"/api/v1/pvc/claim5-2","uid":"uid5-2","creationTimestamp":null
		
		A: ,"annotations":{"pv.kubernetes.io/bind-completed":"yes","pv.kubernetes.io/bound-by-controller":"yes"}},"spec":{"accessModes":["ReadWriteOnce","ReadOnlyMany"],"resources":{"requests":{"storage":"1Gi"}},"volumeName":"volume5-2"},"status":{"phase":"Bound"}}}
		
		B: },"spec":{"accessModes":["ReadWriteOnce","ReadOnlyMany"],"resources":{"requests":{"storage":"1Gi"}}},"status":{"phase":"Pending"}}}
		
	framework_test.go:715: Test "5-2 - complete bind": Volume check failed [A-expected, B-got]: {"volume5-2":{"metadata":{"name":"volume5-2","creationTimestamp":null
		
		A: ,"annotations":{"pv.kubernetes.io/bound-by-controller":"yes"}},"spec":{"capacity":{"storage":"10Gi"},"gcePersistentDisk":{"pdName":""},"accessModes":["ReadWriteOnce","ReadOnlyMany"],"claimRef":{"kind":"PersistentVolumeClaim","namespace":"default","name":"claim5-2","uid":"uid5-2","apiVersion":"v1"},"persistentVolumeReclaimPolicy":"Retain"},"status":{"phase":"Bound"}}}
		
		B: },"spec":{"capacity":{"storage":"10Gi"},"gcePersistentDisk":{"pdName":""},"accessModes":["ReadWriteOnce","ReadOnlyMany"],"persistentVolumeReclaimPolicy":"Retain"},"status":{"phase":"Available"}}}
```

Today I saw a very similar error to the one reported in #26067, but seeing as that was closed as a dupe of this, adding details here:

https://github.com/kubernetes/kubernetes/pull/26105#issuecomment-221321773

https://console.cloud.google.com/storage/browser/kubernetes-jenkins/pr-logs/pull/26105/kubernetes-pull-test-unit-integration/27563/
```
E0523 22:12:02.268570   29064 controller_base.go:259] PersistentVolumeController could not update claim "default/fake-pvc": persistentvolumeclaims "fake-pvc" not found
panic: test timed out after 10m0s

goroutine 19857 [running]:
panic(0x1b5e200, 0xc82a5a1b10)
    /usr/local/go/src/runtime/panic.go:481 +0x3e6
testing.startAlarm.func1()
    /usr/local/go/src/testing/testing.go:725 +0x14b
created by time.goFunc
    /usr/local/go/src/time/sleep.go:129 +0x3a

goroutine 1 [chan receive, 7 minutes]:
testing.RunTests(0x2940538, 0x3567fa0, 0x32, 0x32, 0xc8202c5501)
    /usr/local/go/src/testing/testing.go:583 +0x8d2
testing.(*M).Run(0xc823baff08, 0xc820172d80)
    /usr/local/go/src/testing/testing.go:515 +0x81
main.main()
    k8s.io/kubernetes/test/integration/_test/_testmain.go:152 +0x117
```

https://pantheon.corp.google.com/storage/browser/kubernetes-jenkins/pr-logs/pull/26105/kubernetes-pull-test-unit-integration/27563/

@quinton-hoole, this looks like something new. Let's keep this issue open for this unit test flake:

```
--- FAIL: TestControllerSync (0.15s)
    framework_test.go:711: Test "5-2 - complete bind": Claim check failed [A-expected, B-got result]: {"claim5-2":{"metadata":{"name":"claim5-2","namespace":"default","selfLink":"/api/v1/pvc/claim5-2","uid":"uid5-2","creationTimestamp":null

        A: ,"annotations":{"pv.kubernetes.io/bind-completed":"yes","pv.kubernetes.io/bound-by-controller":"yes"}},"spec":{"accessModes":["ReadWriteOnce","ReadOnlyMany"],"resources":{"requests":{"storage":"1Gi"}},"volumeName":"volume5-2"},"status":{"phase":"Bound"}}}

        B: },"spec":{"accessModes":["ReadWriteOnce","ReadOnlyMany"],"resources":{"requests":{"storage":"1Gi"}}},"status":{"phase":"Pending"}}}

    framework_test.go:715: Test "5-2 - complete bind": Volume check failed [A-expected, B-got]: {"volume5-2":{"metadata":{"name":"volume5-2","creationTimestamp":null

        A: ,"annotations":{"pv.kubernetes.io/bound-by-controller":"yes"}},"spec":{"capacity":{"storage":"10Gi"},"gcePersistentDisk":{"pdName":""},"accessModes":["ReadWriteOnce","ReadOnlyMany"],"claimRef":{"kind":"PersistentVolumeClaim","namespace":"default","name":"claim5-2","uid":"uid5-2","apiVersion":"v1"},"persistentVolumeReclaimPolicy":"Retain"},"status":{"phase":"Bound"}}}

        B: },"spec":{"capacity":{"storage":"10Gi"},"gcePersistentDisk":{"pdName":""},"accessModes":["ReadWriteOnce","ReadOnlyMany"],"persistentVolumeReclaimPolicy":"Retain"},"status":{"phase":"Available"}}}
```


I filled #26256 for integration test flake you reported.

--- a/pkg/controller/persistentvolume/controller_test.go
+++ b/pkg/controller/persistentvolume/controller_test.go
@@ -33,6 +33,7 @@ import (
 // can't reliably simulate periodic sync of volumes/claims - it would be
 // either very timing-sensitive or slow to wait for real periodic sync.
 func TestControllerSync(t *testing.T) {
+	expectedChanges := []int{1, 4, 1, 1}
 	tests := []controllerTest{
 		// [Unit test set 5] - controller tests.
 		// We test the controller as if
@@ -121,7 +122,7 @@ func TestControllerSync(t *testing.T) {
 		},
 	}
 
-	for _, test := range tests {
+	for ix, test := range tests {
 		glog.V(4).Infof("starting test %q", test.name)
 
 		// Initialize the controller
@@ -148,13 +149,17 @@ func TestControllerSync(t *testing.T) {
 			time.Sleep(10 * time.Millisecond)
 		}
 
+		count := reactor.getChangeCount()
+
 		// Call the tested function
 		err := test.test(ctrl, reactor, test)
 		if err != nil {
 			t.Errorf("Test %q initial test call failed: %v", test.name, err)
 		}
 
-		reactor.waitTest()
+		for reactor.getChangeCount() < count+expectedChanges[ix] {
+			reactor.waitTest()
+		}
 
 		evaluateTestResults(ctrl, reactor, test, t)
 	}

